src/WebJobs.Script.WebHost/Controllers/ExtensionsController.cs

```diff
@@ -115,6 +118,14 @@ public async Task<IActionResult> GetJobs()
             return Ok(result);
         }
 
+        [HttpGet]
+        [Route("admin/host/extensions/config")]
+        [RequiresRunningHost]
+        public IActionResult GetExtensionsConfig([FromServices] IHostJsonManager hostJsonManager)
```

Comment:
We should just have a ScriptHost level HostOptionsProvider service that is then accessed here using the same pattern we use here https://github.com/Azure/azure-functions-host/blob/94e5372cbd4bdae0e8646fe6ec25f28cacec9e79/src/WebJobs.Script.WebHost/Controllers/HostController.cs#L244. That is, IScriptHostManager provides access to the currently running host's service instance.

This will allow you to use ctor DI in HostOptionsProvider naturally by taking an `IEnumerable<IWebJobsExtensionOptionsConfiguration>` as a param and let DI give you all the instances.

Reply:
I've never noticed this approach. Thanks. Done.